test-harness/src/main/java/io/jenkins/plugins/casc/misc/junit/jupiter/AbstractRoundTripTest.java

```diff
@@ -0,0 +1,215 @@
+package io.jenkins.plugins.casc.misc.junit.jupiter;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.htmlunit.HttpMethod.POST;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import io.jenkins.plugins.casc.ConfigurationAsCode;
+import io.jenkins.plugins.casc.ConfiguratorException;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.ConsoleHandler;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+import java.util.logging.SimpleFormatter;
+import org.apache.commons.io.IOUtils;
+import org.htmlunit.WebRequest;
+import org.htmlunit.WebResponse;
+import org.htmlunit.util.NameValuePair;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.junit.jupiter.WithJenkins;
+
+/**
+ * This is the JUnit5 equivalent of {@link io.jenkins.plugins.casc.misc.RoundTripAbstractTest}
+ *
+ * Base test to check a complete test of each plugin configuration. What it makes: 1. Configure the
+ * instance with the {@link #configResource()} implemented. 2. Check it was configured correctly. 3.
+ * Check the configuration is valid via Web UI. 4. Apply the configuration via Web UI. 5. Write the
+ * configuration to $JENKINS_HOME/jenkins.yaml. 6. Restart Jenkins. 7. Check the
+ * {@link #stringInLogExpected()} is set during the restart.
+ *
+ * All the plugin author needs to do is override the methods providing: 1. The resource with the
+ * yaml configuration of the plugin in case they use their own name for the file. 2. A way to
+ * validate the configuration is established. 3. A string that should be present in the logs (casc
+ * logger) that guarantees the config is loaded. Usually a weird text configured.
+ */
+@WithJenkins
+public abstract class AbstractRoundTripTest {
```

Comment:
I used a slightly different name besides a different package to reduce the risk of mix-ups. Not sure if this is the best way, so feedback is welcome.